fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -234,13 +239,18 @@ private boolean insertField(LuceneDocumentFromRecord.DocumentField field, final
         return suggestionAdded;
     }
 
-    private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
-                               byte[] primaryKey) throws IOException {
+    private <M extends Message> void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
+                               @Nonnull FDBIndexableRecord<M> record) throws IOException {
+        byte[] primaryKey = record.getPrimaryKey().pack();
         final IndexWriter newWriter = getOrCreateIndexWriter(state, indexAnalyzer, executor, groupingKey);
         BytesRef ref = new BytesRef(primaryKey);
         Document document = new Document();
         document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
-        document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
+        if (record.getRecordType().getPrimaryKey().isFixedWidth(record.getRecordType().getDescriptor())) {
```

Comment:
`RecordType.getPrimaryKey` will include the grouping keys, if they are also part of the primary key. But it's okay for those the be `null`. It's also slightly more efficient to only encode the primary key relative to the index, that is, without any grouping key prefix(es).